include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -2686,6 +2766,181 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::
   return cell_transfer.run(cell_transfer_test);
 }
 
+
+
+namespace RepartitioningPolicyTools
+{
+  namespace
+  {
+    template <typename MeshType>
+    unsigned int
+    compute_n_coarse_cells(const MeshType &mesh)
+    {
+      types::coarse_cell_id n_coarse_cells = 0;
+
+      for (const auto &cell : mesh.get_triangulation().active_cell_iterators())
+        if (!cell->is_artificial())
+          n_coarse_cells =
+            std::max(n_coarse_cells, cell->id().get_coarse_cell_id());
+
+      return Utilities::MPI::max(n_coarse_cells, mesh.get_communicator()) + 1;
+    }
+
+
+
+    template <typename MeshType>
+    unsigned int
+    compute_n_global_levels(const MeshType &mesh)
+    {
+      return mesh.get_triangulation().n_global_levels();
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    add_indices(const TriaIterator<CellAccessor<dim, spacedim>> &cell,
+                const internal::CellIDTranslator<dim> &cell_id_translator,
+                IndexSet &                             is_fine)
+    {
+      is_fine.add_index(cell_id_translator.translate(cell));
```

Comment:
As noted in #12337, I wonder whether this is a good strategy for 32 bit integers. It seems dangerous at least for meshes that are strongly refined adaptively.

Reply:
In that case one would need to switch to 64 bit integers. But let's keep this topic out this PR, since this how it is done currently also for the global coarsening algorithms.